sw/host/ot_certs/src/template/mod.rs

```diff
@@ -67,22 +69,37 @@ pub struct Certificate {
     pub authority_key_identifier: Value<Vec<u8>>,
     /// X509 certificate's public key identifier.
     pub subject_key_identifier: Value<Vec<u8>>,
-    /// DICE TCB model.
+    /// X509 certificate extensions.
+    pub extensions: Vec<CertificateExtension>,
+    /// X509 certificate's signature.
+    pub signature: Signature,
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, Deserialize, Serialize)]
+#[serde(tag = "type", rename_all = "snake_case")]
+pub enum CertificateExtension {
+    /// DICE TCB extension.
+    DiceTcb(DiceExtension),
+}
+
+/// DICE TCB extension.
+#[derive(Clone, Debug, PartialEq, Eq, Deserialize, Serialize)]
+#[serde(deny_unknown_fields)]
+pub struct DiceExtension {
```

Comment:
Why define a new struct here? can you not use the `DiceTcbInfo` struct in `dice_tcb.rs`?

Reply:
I haven't finished refactoring yet. It cannot be used as-is for now because the structure in `dice_tcb.rs` only contains reference and does not own the data. Also the one in `dice_tcb.rs` is more general, as it contains a few fields that we do not use so I am undecided if I should remove them entirely or not? But eventually those two structure will be merge, probably the one in `dice_tcb` will disappear.

Reply:
sg. I think consolidating at least these two should make things more maintainable.